docs/development/design/watch-api.md

```diff
@@ -0,0 +1,191 @@
+# Nexodus Device Watch API Implementation Notes
+
+## The Problem
+
+In order to connect or disconnect devices from a Nexodus managed organization, the list of all desired devices in the organization need replicated to all devices in the organization.  The `ListDevicesInOrganization` operation provides this list and can be accessed via the following REST call:
+
+`GET /api/organizations/{organization_id}/devices`
+
+The `nexd` agent would need to periodically poll this endpoint to get the current desired state from the API server and reconcile the device configuration to add or remove Wireguard peers endpoints as nessary.
+
+As the number of devices (N) in the organization increases, the the number of periodic API requests to `ListDevicesInOrganization` increases linearly.  The size of the reponse list also increases linearly.  Put those two together and you get the reponse bandwith to the API server increasing exponentially (N requests/polling interval x N device elements in the response).  This will clearly lead to scaling challenges with large organizations.
+
+This also has the problem that desired state changes recorded in the apiserver will not take effect immediatly since the `nexd` agents are likely in the middle of their poll interval, and wont poll the API server for a little while.
```

Comment:
```suggestion
This also has the problem that desired state changes recorded in the apiserver will not take effect immediately since the `nexd` agents are likely in the middle of their poll interval and won't poll the API server for a little while.
```